bin/syncSquadFlags.ts

```diff
@@ -0,0 +1,62 @@
+import fastq from 'fastq';
+import '../src/config';
+import createOrGetConnection from '../src/db';
+import { Post, Source, SquadSource } from '../src/entity';
+import { updateFlagsStatement } from '../src/common';
+
+interface Result {
+  sourceId: string;
+  count: number;
+  views: number;
+  upvotes: number;
+}
+
+const QUEUE_CONCURRENCY = 1;
+
+(async () => {
+  const con = await createOrGetConnection();
+  const stream = await con
+    .getRepository(Post)
+    .createQueryBuilder('p')
+    .select('COUNT(p.*)', 'count')
+    .addSelect('SUM(p.views)', 'views')
+    .addSelect('SUM(p.upvotes)', 'upvotes')
+    .addSelect('s.id', 'sourceId')
+    .innerJoin(SquadSource, 's', 's.id = p."sourceId"')
+    .where(`s.type = 'squad'`)
+    .groupBy('s.id')
+    .stream();
+
+  let insertCount = 0;
+  const insertQueue = fastq.promise(
+    async ({ sourceId, views, upvotes, count }: Result) => {
+      console.log('processing stats for: ', sourceId);
+      await con.getRepository(SquadSource).update(
+        { id: sourceId },
+        {
+          flags: updateFlagsStatement<Source>({
+            totalViews: views,
+            totalUpvotes: upvotes,
+            totalPosts: count,
+          }),
+        },
+      );
+      console.log('stats updated for: ', sourceId);
```

Comment:
Probably want to remove this line it is gonna spam a lot when you run the script. But up to you!

Reply:
Also above log `processing stats..`

Reply:
Agree with that. Replaced it with just a log when initializing, just so we know it is actually starting.